ooniprobe/Model/Database/Result.mm

```diff
@@ -90,8 +83,25 @@ -(NSString*)getLocalizedNetworkType{
     return @"";
 }
 
--(void)addRuntime:(float)value{
-    self.runtime+=value;
+-(Measurement*)getFirstMeasurement{
+    SRKResultSet *measurements = [[[[Measurement query] where:@"result_id = ? AND is_rerun = 0" parameters:@[self]] order:@"start_time"] fetch];
+    if ([measurements count] > 0)
+        return [measurements objectAtIndex:0];
+    return nil;
+}
+
+-(Measurement*)getLastMeasurement{
+    SRKResultSet *measurements = [[[[Measurement query] where:@"result_id = ? AND is_rerun = 0" parameters:@[self]] orderByDescending:@"start_time"] fetch];
+    if ([measurements count] > 0)
+        return [measurements objectAtIndex:0];
+    return nil;
+}
+
+-(float)getRuntime{
+    Measurement *first = [self getFirstMeasurement];
```

Comment:
What happens if you get `nil` here? Will it not trigger a NULL pointer exception? Can we maybe handle that case a bit more gracefully and perhaps set a fixed maxRuntime if that happens?

Reply:
You can't get nil because you can't access the test summary screen if there are no measurement in the result. See https://github.com/ooni/probe/issues/855

Reply:
The fact you cannot have this be `nil` because of where you call the `getRuntime` in the code is something that may change in the future and you are guaranteed to forget to add the safety then.

Please practice good defensive coding and add checks for `nil`. I think it's fine to also just handle it as an exception, but it should be explicited.

Reply:
Would a 
```
    if (first == nil || last == nil)
        return 0;
```
work?

Reply:
What pattern is being used elsewhere for handling null pointer exceptions?

Reply:
We are not using exceptions at all in iOS. They are not commonly used.